src/Aspire.Hosting.Azure/Provisioning/Provisioners/AzureProvisioner.cs

```diff
@@ -242,65 +260,71 @@ private async Task ProcessResourceAsync(IConfiguration configuration, Lazy<Task<
 
             try
             {
                 var provisioningContext = await provisioningContextLazy.Value.ConfigureAwait(false);
 
                 await provisioner.GetOrCreateResourceAsync(
-                    resource,
+                    resource.AzureResource,
                     provisioningContext,
                     cancellationToken).ConfigureAwait(false);
 
-                resource.ProvisioningTaskCompletionSource?.TrySetResult();
+                resource.AzureResource.ProvisioningTaskCompletionSource?.TrySetResult();
+                await PublishConnectionStringAvailableEventAsync().ConfigureAwait(false);
             }
             catch (AzureCliNotOnPathException ex)
             {
                 resourceLogger.LogCritical("Using Azure resources during local development requires the installation of the Azure CLI. See https://aka.ms/dotnet/aspire/azcli for instructions.");
-                resource.ProvisioningTaskCompletionSource?.TrySetException(ex);
+                resource.AzureResource.ProvisioningTaskCompletionSource?.TrySetException(ex);
             }
             catch (MissingConfigurationException ex)
             {
                 resourceLogger.LogCritical("Resource could not be provisioned because Azure subscription, location, and resource group information is missing. See https://aka.ms/dotnet/aspire/azure/provisioning for more details.");
-                resource.ProvisioningTaskCompletionSource?.TrySetException(ex);
+                resource.AzureResource.ProvisioningTaskCompletionSource?.TrySetException(ex);
             }
             catch (JsonException ex)
             {
-                resourceLogger.LogError(ex, "Error provisioning {ResourceName} because user secrets file is not well-formed JSON.", resource.Name);
-                resource.ProvisioningTaskCompletionSource?.TrySetException(ex);
+                resourceLogger.LogError(ex, "Error provisioning {ResourceName} because user secrets file is not well-formed JSON.", resource.AzureResource.Name);
+                resource.AzureResource.ProvisioningTaskCompletionSource?.TrySetException(ex);
             }
             catch (Exception ex)
             {
-                resourceLogger.LogError(ex, "Error provisioning {ResourceName}.", resource.Name);
-
-                resource.ProvisioningTaskCompletionSource?.TrySetException(new InvalidOperationException($"Unable to resolve references from {resource.Name}"));
+                resourceLogger.LogError(ex, "Error provisioning {ResourceName}.", resource.AzureResource.Name);
+                resource.AzureResource.ProvisioningTaskCompletionSource?.TrySetException(new InvalidOperationException($"Unable to resolve references from {resource.AzureResource.Name}"));
             }
         }
+
+        async Task PublishConnectionStringAvailableEventAsync()
+        {
+            var connectionStringAvailableEvent = new ConnectionStringAvailableEvent(resource.Resource, serviceProvider);
```

Comment:
Do we need to publish for child resources?

Reply:
So what I've done here is I've made it so that when the server gets its connection string event fired, it then fires its child resource connection string events. This is a decision we would need to make on a per resource basis but I think it should hold for most resource types.